hector_mapping/src/HectorMappingRos.cpp

```diff
@@ -124,6 +124,19 @@ HectorMappingRos::HectorMappingRos()
     odometryPublisher_ = node_.advertise<nav_msgs::Odometry>("scanmatch_odom", 50);
   }
 
+  // Cache lidar static transforms
+  std::vector<std::string> static_laser_frame_ids;
+  private_nh_.getParam("static_laser_frame_ids", static_laser_frame_ids);
+  for (const auto& laser_frame : static_laser_frame_ids)
+  {
+    tf::StampedTransform tf_base_to_lidar_frame;
+    if (util::LookupTransform(p_base_frame_, laser_frame, &tf_base_to_lidar_frame))
+    {
+      ROS_INFO("HectorSM Transform from \"%s\" to \"%s\" has been cached", p_base_frame_.c_str(), laser_frame.c_str());
+      cached_static_lidar_transforms_[laser_frame] = tf_base_to_lidar_frame;
+    }
+  }
```

Comment:
iterate over all static frame ids and cache their transforms from `base_frame`.

Reply:
`cached_static_lidar_transforms_` is a hash table of transforms, allowing the lookup to be O(1) in average